CompilerSource/ipc/Path.fbs

```diff
@@ -0,0 +1,23 @@
+
+struct PathPoint {
+  x:int;
+  y:int;
+  speed:int;
+}
+
+table Path {
+  name:string;
+  id:int;

```

Comment:
Ok, so now that I have added another resource type, this is an important question to ask. Because neither FlatBuffers nor protobuf have inheritance, what should we do about common fields like `name` and `id`?

* We can obviously just write them out for each resource, I don't care.
* We could make the resource types a union.
* We could use composition and store those fields in a struct referenced by all resources.

Reply:
I don't have a strong opinion on this; the two are related, sure, but that's basically all there is to identifying something. Maybe later we might want to add a creation timestamp? If we decided to do that, then having this separate message type would pay off. Otherwise, it's a bit pointless. Do whichever you feel looks best.

Reply:
No, that's not what I'm getting at actually. I meant how Path and Background both have name, and for each new resource type I must type `name` and similarly for `id`.